### ISSUE TITLE
Feature request: allow center= to accept more than a boolean

### ISSUE THREAD

**Issue by cbrinker** [openscad/openscad#265]
I find myself spending a lot of my modeling time doing something akin to this:

  translate([0,0,10/2]) cylinder(r=5, h=10, center=true); 

Which suffers if I need to adjust the height of the cylinder to 20 I have to remember to update the translate. It would be nice to accomplish the same with something akin to:

  cylinder(r=5, h=10, center=[1,1,0])

Does this already exist and I didn't catch it in the docs?

Thanks,
Chris
##

**Comment by Salamandar** [comment]
Up ;)
I was gonna open an issue when I saw this one. This would considerably simplify a LOT of openSCAD files.

**Comment by ranguli** [comment]
@kintel Should we move this to the [ideas for development](https://github.com/openscad/openscad/wiki/Ideas-for-Development-Tasks) section of the wiki if it's under consideration, or close the issue otherwise?

**Comment by kintel** [comment]
@ranguli A writeup on the wiki would indeed be appreciated. Not sure if that warrants closing the issue as the discussion here may be worth keeping, but at least it would anchor the feature request there.

**Comment by ranguli** [comment]
@kintel Perhaps a "Long-term" or "Reviewed" label might help to clean up some of the older issues that have been examined and aren't ready to be closed but aren't currently being worked on?

**Comment by HWXLR8** [comment]
I will pay cold hard cash to see this implemented. Name your price.

**Comment by jordanbrown0** [comment]
See https://github.com/BelfrySCAD/BOSL2/wiki/attachments.scad for a library that does this and much more.

**Comment by pipatron** [comment]
> See https://github.com/BelfrySCAD/BOSL2/wiki/attachments.scad for a library that does this and much more.

Agreed, can't imagine doing anything complex in OpenSCAD without BOSL2. Looking at the speed of which new features are implemented into core OpenSCAD, this won't happen in the language itself.

**Comment by gsohler** [comment]
Yes,  BOSL2 is a great library, but it's based on a limited tool., So BOSL2
is , too
OpenSCAD would be so much better if it provided access to the generated
vertices, But it doesn't.

BOSL2 made it possible to circumvent  this limitation by introducing VNF.
VNF is basically vertex and triangle information of your solid. but you can
only have it when created by BOSL2, you would never get from an
arbitrary OpenSCAD solid.
I dare to say, that my fork is moving way faster. You can easily access the
vertices of any solid, you can have 1D, 2D or 3D objects and you can store
arbitrary data with any solid.

Generally  I dont understand, why openscad guys always suggest alternative
solutions to  a request rather than addressing the issue directly.
my 2 cents ...

**Comment by GilesBathgate** [comment]
The comments seem to have gone a bit off-topic. While work arounds like BSOL2 exist, the OP seems to be a reasonable request, and could probably be resolved more generally by the [align() proposal #1610](https://github.com/openscad/openscad/issues/1610)  . I implemented such a module in RapCAD, and it works well. It utilises a bounding box of the final geometry, so I frequently use it in the scenario where I don't precisely know the dimentions (importing an STL) . I think it would work suitablly for the OP's intended use case.
